packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.tsx

```diff
@@ -99,6 +100,17 @@ export function ComboboxContent(props: ComboboxContentProps): JSX.Element {
         setSearchValue={setSearchValue}
       />
 
+      {multiselect && (
+        <ComboboxHeader
+          subjectNoun={props.subjectNoun}
+          selectedCount={selectedOptions.length}
+          onClearAll={() => {
+            setInternalSelected([]);
```

Comment:
just a heads up, this won't work for the case where we were given `onSelect`

the internal state is only used for `onClose` where the passed in state of `selected` can diverge from the internal state

for `onSelect` we'll need to use the function we were given. you can see an example of how this work in `handleSelection` where we either use the `props.onSelect` or the `setInternalState` depending on if it's an `onClose` or not